mirrord/agent/src/steal/connection.rs

```diff
@@ -581,3 +659,152 @@ impl TcpConnectionStealer {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::net::SocketAddr;
+
+    use bytes::Bytes;
+    use futures::{future::BoxFuture, FutureExt};
+    use http::{Method, Request, Response, Version};
+    use http_body_util::{Empty, StreamBody};
+    use hyper::{
+        body::{Frame, Incoming},
+        service::Service,
+    };
+    use hyper_util::rt::TokioIo;
+    use mirrord_protocol::tcp::{ChunkedRequest, DaemonTcp, InternalHttpBodyFrame};
+    use tokio::{
+        net::{TcpListener, TcpStream},
+        sync::{
+            mpsc::{self, Receiver, Sender},
+            oneshot,
+        },
+    };
+    use tokio_stream::wrappers::ReceiverStream;
+    use x509_parser::nom::AsBytes;
```

Comment:
Whaat happened here :laughing: 
Maybe let's not use this `nom` trait, this is confusing. Instead of `b"string".as_bytes().into()` you can do `b"string".to_vec().into()`